flows/engine/engine.go

```diff
@@ -143,24 +139,30 @@ func resumeNode(run flows.FlowRun, node flows.Node, step flows.Step, event flows
 		return noDestination, step, nil
 	}
 
-	err := wait.End(run, step, event)
+	// try to resume our wait with the first caller event
+	err := wait.End(run, step, callerEvents[0])
```

Comment:
Ya, I can't currently really think of a case when we would resume with more than one event so I'm fine with it as is for now.